account/recovery.go

```diff
@@ -220,16 +379,121 @@ func recoverInitialState(ctx context.Context, cfg RecoveryConfig) ([]*Account,
 func updateAccountStates(cfg RecoveryConfig, client *rpcclient.Client,
 	accounts []*Account) ([]*Account, error) {
 
-	// TODO (positiveblue): update account states
+	for i := cfg.FirstBlock; i < uint32(cfg.CurrentBlockHeight); i++ {
+		blockHeight := i
+
+		blockHash, err := client.GetBlockHash(int64(i))
+		if err != nil {
+			return nil, err
+		}
+
+		block, err := client.GetBlock(blockHash)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, tx := range block.Transactions {
+			for idx, acc := range accounts {
+				// It's safe to ignore accounts that are in a
+				// terminal state.
+				if acc.State == StateClosed {
+					continue
+				}
+				found := poolscript.IncludesPreviousOutPoint(
+					tx, acc.OutPoint,
+				)
+
+				if !found {
+					continue
+				}
+
+				newAcc, err := findAccountUpdate(
+					cfg, acc, blockHeight, tx,
+				)
+				if err != nil {
+					log.Debugf("unable to find account "+
```

Comment:
I think if we can't find an account update, we need to consider the account to be closed. Perhaps we should update this log to info and say something like `Account was spent in tx %x but not re-created. Assuming account was fully spent or closed.` and then set the state to closed (also see comment below about the dust value).